framework/services/bookService.js

```diff
@@ -0,0 +1,38 @@
+import { apiClient } from './apiClient.js';
+
+export async function createBook({ userId, collectionOfIsbns, username, password }) {
+  try {
+    const basicAuth = Buffer.from(`${username}:${password}`).toString('base64');
+
+    const headers = {
+      Authorization: `Basic ${basicAuth}`
+    };
+
+    const response = await apiClient.post('/BookStore/v1/Books', { userId, collectionOfIsbns }, { headers });
```

Comment:
В axios  можно  Basic auth по другому сделать, без явной кодировки данных

```js
// `auth` indicates that HTTP Basic auth should be used, and supplies credentials.
  // This will set an `Authorization` header, overwriting any existing
  // `Authorization` custom headers you have set using `headers`.
  // Please note that only HTTP Basic auth is configurable through this parameter.
  // For Bearer tokens and such, use `Authorization` custom headers instead.
  auth: {
    username: 'janedoe',
    password: 's00pers3cret'
  },
  ```
  
  https://axios-http.com/docs/req_config